Documentation/CRDs/Object-Storage/ceph-object-store-crd.md

```diff
@@ -147,6 +149,8 @@ gateway:
 This will create a service with the endpoint `192.168.39.182` on port `80`, pointing to the Ceph object external gateway.
 All the other settings from the gateway section will be ignored, except for `securePort`.
 
+* `hostnames`: A list of hostnames to use for accessing the bucket directly like a subdomain in the endpoint. These hostnames will be added to the zonegroup's hostname list and objectstore can be accessed only by the hostname from the list. By default RGW will add the service endpoint as the hostname and feature is disabled. If the feature is enabled, users need to add the all the endpoints like the openshift route, ingress loadbalancer, TLS cert endpoints etc. to the list. The hostname list can be updated at any time by updating the object store CRD.
```

Comment:
A design doc would typically be an update to the /design folder, rather than updating the user documentation. The docs should not be updated until the feature is implemented. Perhaps this means a new section in [zone.md](https://github.com/rook/rook/blob/master/design/ceph/object/zone.md)?

I'd expect a few questions to be answered in the design:
- Summarize the scenario. Why would users want the feature?
- How is it implemented? Does Rook need to call a radosgw-admin command to apply the hostnames?
- If TLS is enabled, does the cert need certain names or alternate names? 
- Does multi-site have any special considerations? Or it works independently of multi-site?
- Any special test considerations?
- Other answers to questions in the discussion of #4780. 
- - How does this relate to the `customEndpoints` property?

Reply:
I'd also like to see a workflow for how this could be enabled in Rook today. E.g., stop the rook operator, modify rgw deployment, set ceph configs, create a new service, etc.

This also really helps solidify what the design needs to do practically, which helps when implementing and reviewing.